File: simple-stack-example-multistack/src/main/java/com/zhuinden/simplestackdemomultistack/presentation/paths/main/mail/MailKey.java
package com.zhuinden.simplestackdemomultistack.presentation.paths.main.mail;

import android.os.Parcelable;

import com.google.auto.value.AutoValue;
import com.zhuinden.simplestackdemomultistack.R;
import com.zhuinden.simplestackdemomultistack.application.Key;
import com.zhuinden.simplestackdemomultistack.application.MainActivity;

/**
 * Created by Zhuinden on 2017.02.19..
 */
@AutoValue
public abstract class MailKey
        extends Key {
    @Override
    public int layout() {
        return R.layout.path_mail;
    }

    public static Parcelable create() {
        return new AutoValue_MailKey();
    }

    @Override
    public String stackIdentifier() {
        return MainActivity.StackType.MAIL.name();
    }
}

File: samples/legacy-samples/simple-stack-example-multistack-view/src/main/java/com/zhuinden/simplestackdemomultistack/features/main/mail/MailKey.kt
package com.zhuinden.simplestackdemomultistack.features.main.mail

import com.zhuinden.simplestackdemomultistack.R
import com.zhuinden.simplestackdemomultistack.application.MainActivity
import com.zhuinden.simplestackdemomultistack.core.navigation.MultistackViewKey
import kotlinx.parcelize.Parcelize

@Parcelize
data object MailKey : MultistackViewKey() {
    override fun layout(): Int = R.layout.mail_view
    override fun stackIdentifier(): String = MainActivity.StackType.MAIL.name
}
